DEVDOCS-4429: [new] Headless Carts, add locale info; [update] Customer Login (#1631)
bigcommerce/dev-docs: docs/api-docs/headless/carts.md

In this section, we will explain how to use the Carts API to create and manage carts. Additionally, we will discuss how to redirect shoppers from a headless storefront to the BigCommerce hosted cart and checkout pages.

<!-- theme: info -->
> #### Locale support
> The Carts API supports selling in different markets by allowing locale-based overrides for product details. Supply a cart's locale and add `option_selections` to the cart's line items to configure alternative product names, option names, and modifier values. The Carts API stores the locale and makes it available to the Checkout and Orders APIs. 

## Creating a cart

The [Carts API](/api-reference/store-management/carts) lets you create carts for both existing and guest customers. 

> To create a cart, send a request to the [Create a Cart](/api-reference/store-management/carts/cart/createacart) endpoint.

Reviewer: ```suggestion
To create a cart, send a request to the [Create a cart](/api-reference/store-management/carts/cart/createacart) endpoint.
```

Reviewer: done locally

In this section, we will explain how to use the Carts API to create and manage carts. Additionally, we will discuss how to redirect shoppers from a headless storefront to the BigCommerce hosted cart and checkout pages.

<!-- theme: info -->

> > #### Note

Reviewer: ```suggestion
> #### Locale support
```

> > The Carts API supports selling in different countries by allowing overrides for locale and product details. Merchants can supply a cart's locale, alternative product names, and option values.  

Reviewer: ```suggestion
> The Carts API supports selling in different markets by allowing locale-based overrides. To use alternative product names and option values, <specify those values on the product object and then add the corresponding locale to a cart>.  
```

@bc-traciporter I've added the carrots around approximate text that captures this gist. I leave the actual writing to you, as well the fact checking if this seems wrong or doesn't make sense. I am not 100% about whether option values are pre-existing on the product object, or defined ad hoc in the cart. If they're ad hoc, the gist could be something like this:

```suggestion
> The Carts API supports selling in different markets by allowing locale-based overrides. To use alternative product details, <use the product object to specify product info for a given locale, then add the corresponding locale to a cart. Your application can use the locale to set its own option values>.  
```

Reviewer: never mind, this is wrong!

            "product_id": 80,
            "variant_id": 64
        }
    ],
    "locale": "en-us"
}
```
To create a cart with option selections, include `option_id` and `option_value` in your `POST` request.

> ```json

Reviewer 2: ```suggestion
```json lineNumbers
```

Reviewer 2: @bc-traciporter I noticed that the other code blocks don't have "lineNumbers," so if you commit my suggestion, there are others also that need to be changed

Author: @bc-andreadao what does lineNumbers do? Can you tell me why it should be added?

Reviewer 2: It adds lineNumbers for your code blocks

Reviewer 2: ![Uploading linenumbers.png…]()

It is possible to generate a redirect URL when creating a cart using the [Create a Cart](/api-reference/store-management/carts/cart/createacart) endpoint by appending the `include=redirect_urls` query parameter to the request URL.

```http title="Create a cart"
POST https://api.bigcommerce.com/stores/{{store_hash}}/v3/carts?include=redirect_urls
Accept: application/json
Content-Type: application/json
X-Auth-Token: {{ACCESS_TOKEN}}

{
  // request body
}

```

### Signing in and redirecting a customer

> 

Reviewer: @bc-traciporter Let's revise this section to point to our dedicated documentation on the Customer Login API.
